lib/Optimized64/KeccakP-1600-opt64.c

```diff
@@ -1003,24 +1003,33 @@ void KeccakP1600_opt64_ExtractBytes(const void *state, unsigned char *data, unsi
 size_t KeccakP1600_opt64_12rounds_FastLoop_Absorb(void *state, unsigned int laneCount, const unsigned char *data, size_t dataByteLen)
 {
     size_t originalDataByteLen = dataByteLen;
+
     declareABCDE
     #ifndef KeccakP1600_opt64_fullUnrolling
     unsigned int i;
     #endif
     uint64_t *stateAsLanes = (uint64_t*)state;
     uint64_t *inDataAsLanes = (uint64_t*)data;
 
-    assert(laneCount == 21);
+    assert(laneCount == 21 || laneCount == 17);
 
-    #define laneCount 21
-    copyFromState(A, stateAsLanes)
-    while(dataByteLen >= laneCount*8) {
-        addInput(A, inDataAsLanes, laneCount)
-        rounds12
-        inDataAsLanes += laneCount;
-        dataByteLen -= laneCount*8;
+    if (laneCount == 21) {
+        copyFromState(A, stateAsLanes)
+        while(dataByteLen >= laneCount*8) {
+            addInput(A, inDataAsLanes, laneCount)
+            rounds12
+            inDataAsLanes += laneCount;
+            dataByteLen -= laneCount*8;
+        }
+        copyToState(stateAsLanes, A)
+    } else if (laneCount == 17) {
+        while(dataByteLen >= laneCount*8) {
```

Comment:
Here again, how difficult would it be to transpose the method for 21 lanes to 17 lanes?